stl/inc/chrono

```diff
@@ -2209,6 +2221,2949 @@ namespace chrono {
+        static constexpr bool is_steady = false;
+
+        _NODISCARD static time_point now() noexcept { // get current time; undo epoch adjustment
+            return time_point(duration(_Xtime_get_ticks() + __std_fs_file_time_epoch_adjustment)); // TRANSITION, ABI
+        }
+
+#if _HAS_CXX20
+        // Assumes that FILETIME counts leap seconds only after the first 27 (i.e., after 1 January 2017), even though
+        // systems can opt out of this behavior.
+        static constexpr chrono::seconds _Skipped_filetime_leap_seconds{27};
+        static constexpr chrono::sys_days _Cutoff{
+            chrono::year_month_day{chrono::year{2017}, chrono::January, chrono::day{1}}};
+
+        template <class _Duration>
+        _NODISCARD static chrono::utc_time<common_type_t<_Duration, chrono::seconds>> to_utc(
+            const chrono::file_time<_Duration>& _File_time) {
+            using namespace chrono;
+            using _CommonType = common_type_t<_Duration, seconds>;
+            const auto _Ticks = _File_time.time_since_epoch()
+                              - _CHRONO duration_cast<seconds>(duration{__std_fs_file_time_epoch_adjustment});
+
+            if (_Ticks < _Cutoff.time_since_epoch()) {
+                return utc_clock::from_sys(sys_time<_CommonType>{_Ticks});
+            } else {
+                return utc_time<_CommonType>{_Ticks + _Skipped_filetime_leap_seconds};
+            }
+        }
+
+        template <class _Duration>
+        _NODISCARD static chrono::file_time<common_type_t<_Duration, chrono::seconds>> from_utc(
+            const chrono::utc_time<_Duration>& _Utc_time) {
+            using namespace chrono;
+            file_time<common_type_t<_Duration, seconds>> _File_time{
+                _CHRONO duration_cast<seconds>(duration{__std_fs_file_time_epoch_adjustment})};
+
+            if (_Utc_time < utc_seconds{_Cutoff.time_since_epoch()} + _Skipped_filetime_leap_seconds) {
+                _File_time += utc_clock::to_sys(_Utc_time).time_since_epoch();
+            } else {
+                _File_time += _Utc_time.time_since_epoch() - _Skipped_filetime_leap_seconds;
+            }
+
+            return _File_time;
+        }
+#endif // ^^^ _HAS_CXX20
+    };
+} // namespace filesystem
+#endif // ^^^ _HAS_CXX17
+
+namespace chrono {
+#if _HAS_CXX20
+    // [time.clock.conv]
+
+    // STRUCT TEMPLATE clock_time_conversion
+    template <class _DestClock, class _SourceClock>
+    struct clock_time_conversion {};
+
+    // [time.clock.cast.id]
+
+    template <class _Clock>
+    struct clock_time_conversion<_Clock, _Clock> {
+        template <class _Duration>
+        _NODISCARD time_point<_Clock, _Duration> operator()(const time_point<_Clock, _Duration>& _Time) const
+            noexcept(is_arithmetic_v<typename _Duration::rep>) /* strengthened */ {
+            return _Time;
+        }
+    };
+
+    template <>
+    struct clock_time_conversion<system_clock, system_clock> {

```

Comment:
These are required to avoid ambiguity given the entire system of specializations.

We want the behavior of `clock_time_conversion<_Clock, _Clock>`, but the presence of `clock_time_conversion<system_clock, _SourceClock>` and `clock_time_conversion<_DestClock, system_clock>` introduces ambiguity. (Each can match `<system_clock, system_clock>` in isolation, but none is more specialized than the other: https://godbolt.org/z/rKP8r1jha ).

Reply:
Oh, of course -thanks.